### DISCUSSION TITLE
Use Gitter for realtime collaboration/support

### DISCUSSION THREAD

**Discussion by dmb2** [stumpwm/stumpwm#1110] [Ideas]
Currently #stumpwm on freenode is lightly monitored.  When users come looking for support they rarely get real time responses, and new community members may be put off by not having easy access to support.

Proposal: Use gitter with github integration for real time chat and support.  For users who want to use IRC, use irc.gitter.im and login with a gitter provided token.

**Comment by sdilts** [comment]
I've used gitter for several other projects, and I find it to be really
useful, especially its ability to cross-reference pull requests and issues.

**Comment by m-g-r** [comment]
I just started to use IRC again last month (with a revived beirc) and joined #stumpwm on Freenode as well. But you split it up in Gitter and #stumpwm now or what is the state?  I am reluctant to use Gitter I have to admit.

**Comment by tychoish** [comment]
I like #stumpwm on Freenode, and tend to camp there, and would be down to have more conversations about development (it tends to be user support focused.)

**Comment by emacsomancer** [comment]
I'd also vote instead for increase presence on #stumpwm on freenode

**Comment by PuercoPop** [comment]
I haven't been logging in the past year or so, but afair @jorams is frequently there and so are some contributors. Having a gitter with persistent history makes is easier for conversations to occur with a longer delay so it doesn't hurt afaict (I know #stumpwm is logged but reading the logs to see if someone replied is really good UX).

**Comment by tychoish** [comment]
I wrote up a proposal (and created a demo telegram threaded-group) for conversation (mostly perhaps for development conversation and less for user support, though both are important) as I am somewhat allergic to yet-another-chat-app.

**Comment by dmb2** [comment]
What is the objection to irc.gitter.im? 

I (sheepishly) do not know how to idle in irc and have it ping me when there's activity.  I think it felt slightly easier to hop in and out of gitter and catch up on what I missed.  Really, I'd like a service where people can have realtime converstations, but if they need to ping me to hop in they can and I'll get a notification somehow. If there's a way to do that in IRC that's easy to set up I'm game to reconsolidate things on freenode.

**Reply by tychoish** [reply]
> I (sheepishly) do not know how to idle in irc and have it ping me when there's activity.

In my IRC days I ran a bouncer and had local clients connected to the bouncer which gave me local desktop notifications (often pushed from emacs or something running in a screen/tmux session in different stumpgroup or whatever), so that feels like a solved problem. I know people who used IRCloud as a service to do something similar (that went to the phone) and it's great, but doesn't manage to make it _not IRC_ at the same time. 

> What is the objection to irc.gitter.im?

To be honest, after basically 20 years, I sort of burnt out on it, and while I managed to (for many years) to sort of make IRC the lingua-franca protocol for my life (with bitlbee) but lots of things have happened in the last few years (Slack, facebook Messenger is less compatible, discord, text messaging, encrypted services have to follow your phone, etc.) and the experience isn't great. It's also kind of a crappy protocol by modern standards.

------

There is no right answer, I sort of describe our current era as "the era of chaos messaging" and sort of just do whatever. My preferences (telegram) are mostly because it shows up on my phone well, works on the desktop, the emacs package is great, writing bots against their APIs and libraries is pretty nice, and I've managed to convince a lot of people to go with that so it works. I do talk to other groups of people in other protocols and I sort of feel like you have to meet people where they're at. 

Having said that, my preference is for services that don't require a dedicated tab or window to be open, which feels like a bigger context switch than just seeing a group in a list of your other messages (one of the things I end up liking the most about telegram groups, is they have a "threaded" format, where you can have sub-channels within a group thread. I think whatsapp has a similar concept.)

Anyway, if there are people using it then great, if not, other things may be good too.
